### ISSUE TITLE
Nextjs production with lib error in V12

### ISSUE THREAD

**Issue by VojtechVidra** [nrwl/nx#5312]
## Current Behavior

Nextjs app in production build which is using component from generated library is throwing error. 

## Expected Behavior

In v11 this works just fine.

## Steps to Reproduce

I created simple repo which is reproducing this. Build and run the app in prod env. Open it in a browser and there should be an error. More info in Readme of the repo.

https://github.com/VojtechVidra/nx-12-lib-nextjs-issue

### Failure Logs

Error from browser console:
```
framework.744d8ce69139642428f6.js:1 TypeError: Object(...) is not a function
    at c (index-6f312c2d7a058d85cfa9.js:1)
    at ro (framework.744d8ce69139642428f6.js:1)
    at Hu (framework.744d8ce69139642428f6.js:1)
    at Pi (framework.744d8ce69139642428f6.js:1)
    at xi (framework.744d8ce69139642428f6.js:1)
    at _i (framework.744d8ce69139642428f6.js:1)
    at vi (framework.744d8ce69139642428f6.js:1)
    at fi (framework.744d8ce69139642428f6.js:1)
    at Gi (framework.744d8ce69139642428f6.js:1)
    at framework.744d8ce69139642428f6.js:1
```

### Environment

yarn run v1.22.10
$ nx report

>  NX  Report complete - copy this into the issue template

  Node : 15.11.0
  OS   : darwin arm64
  yarn : 1.22.10
  
  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 12.0.1
  @nrwl/cypress : 12.0.1
  @nrwl/devkit : 12.0.1
  @nrwl/eslint-plugin-nx : 12.0.1
  @nrwl/express : Not Found
  @nrwl/jest : 12.0.1
  @nrwl/linter : 12.0.1
  @nrwl/nest : Not Found
  @nrwl/next : 12.0.1
  @nrwl/node : Not Found
  @nrwl/react : 12.0.1
  @nrwl/schematics : Not Found
  @nrwl/tao : 12.0.1
  @nrwl/web : 12.0.1
  @nrwl/workspace : 12.0.1
  @nrwl/storybook : 12.0.1
  @nrwl/gatsby : Not Found
  typescript : 4.2.4

**Comment by aiji42** [comment]
This may be a duplicate of this issue.
https://github.com/nrwl/nx/issues/5323#issuecomment-818428669

**Comment by aiji42** [comment]
I had the same problem. The problem was solved by setting `NODE_ENV=production` when executing the command.

```
NODE_ENV=production yarn nx run my-app:build:production
```

**Comment by aiji42** [comment]
There seems to be a bug that the environment variable becomes empty when executing the build command in production mode.

**Comment by VojtechVidra** [comment]
Nice, after building with explicit `NODE_ENV=production` it works for me as well. Thanks for the tip.

**Comment by kirjai** [comment]
Should hopefully be fixed in the next reelase by #5574
